doc/src/manual/performance-tips.md

```diff
@@ -477,6 +477,14 @@ Here, we happened to know that the first element of `a` would be an [`Int32`](@r
 an annotation like this has the added benefit that it will raise a run-time error if the
 value is not of the expected type, potentially catching certain bugs earlier.
 
+In the case that the type of `a[1]` is not known precisely, `x` can be declared via
+`x = convert(Int32,a[1])::Int32`.  The
+use of the `convert` function allows `a[1]` to be any object convertible to an `Int32` (such as `UInt8`), thus
+increasing the genericity of the code by loosening the type requirement.
+Notice that the `convert` function itself needs a type annotation in this context in order to achieve type stability.
+This is because the compiler cannot deduce the type of the return value of a function, even
+`convert`, unless the types of all its arguments are known.
```

Comment:
Perhaps "...the types of all the arguments...", since "its" could refer to the compiler in this sentence?